### PR TITLE
Count and exists pair functions

### PR DESCRIPTION
**Pull request by j4mie**
As discussed in https://github.com/dabapps/django-rest-framework-serialization-spec/pull/68#issuecomment-836290871, this adds `pairs.count` and `pairs.has`.

### CODE REVIEW

File: django_readers/pairs.py [hunk 1]
```diff
@@ -42,6 +43,24 @@ def field_display(name):
...
+        projectors.attr(attr_name),
+    )
+
+
+def exists(name, distinct=True):
+    attr_name = f"{name}_count"
+    return (
+        qs.annotate(**{attr_name: Count(name, distinct=distinct)}),
+        projectors.alias(
+            f"{name}_exists", projectors.attr(attr_name, transform_value=bool)
```
Comment (pmg103) [inline django_readers/pairs.py:59]: `name` would usually be plural wouldn't it? `cats_exists` is a slightly odd naming scheme. `f"{name}_exist"` would give `cats_exist` which is more natural perhaps? but then it differs from `exists()` and `Exists()` by only one character which is a recipe for confusion.

Usually when doing this manually I call the key `has_cats` so how about `f"has_{name}"`?

**Comment by j4mie** [comment]
It is a bit odd @pmg103 - my test models are like this:

```
class Owner(models.Model):
    name = models.CharField(max_length=100)

class Widget(models.Model):
    name = models.CharField(max_length=100, null=True)
    owner = models.ForeignKey(Owner, null=True, on_delete=models.SET_NULL)
```

So `owner` would have a `widget_set` reverse relationship - but you still have to do `Owner.objects.annotate(Count("widget"))` (rather than `Count("widget_set")`). If you try the latter you get 

```
django.core.exceptions.FieldError: Cannot resolve keyword 'widget_set' into field. Choices are: group, group_id, id, name, widget
```

I'm not sure anything really works for every case - `widget_exists` works but `widgets_exists` doesn't, `has_widget` makes sense but isn't actually the correct pluralisation because it's a one-to-many relationship, so should be `has_widgets` or `has_widget_set`.

**Comment by j4mie** [comment]
I guess maybe `has_widget` is better? I'd want to call the pair function `has` though in that case, I think it might get a bit confusing otherwise.

**Comment by j4mie** [comment]
I've renamed `exists` to `has`. Definitely better.

**Review by pmg103 — APPROVED** [review]

**Pull request merged**